daemon/openrazer_daemon/hardware/keyboards.py

```diff
@@ -1419,3 +1419,22 @@ class RazerBladePro2019(_RippleKeyboard):
                'set_ripple_effect', 'set_ripple_effect_random_colour']
 
     DEVICE_IMAGE = "https://assets2.razerzone.com/images/razer-blade-pro-17/razer-blade-pro-17-2019-OGimage-1200x630.jpg"
+
+
+class RazerBlade2020Advanced(_RippleKeyboard):
+    """
+    Class for the Razer Blade 15 (2020) Advanced
+    """
+    EVENT_FILE_REGEX = re.compile(r'.*Razer_Blade(-if01)?-event-kbd')
+
+    USB_VID = 0x1532
+    USB_PID = 0x0253
+    HAS_MATRIX = True
+    MATRIX_DIMS = [6, 15]
```

Comment:
Probably `[6, 16]` instead?

Reply:
I counted the keys and there are 6 rows and 15 columns - see the screenshot of the kbd in #1201

Reply:
I'm not sure why the others use 16 either though - maybe I'm misunderstanding what this number means?

Reply:
Does `MATRIX_DIMS` refer to the keyboard backlights rather than the keys? 

I looked at my Blackwidow Ultimate but am still unsure of where the extra column is coming from. The layout seems to be 6 keys vertical and up to 21 keys horizontal, whereas the code here has `MATRIX_DIMS = [6, 22]` .

Reply:
Each key except the space bar on the Blade 15 advanced has its own backlight, so it shouldn't matter if it is referring to backlights or keys

Reply:
Turned out it needed to be 16, otherwise the ripple effect doesn't reach all the way to the right, even though the keyboard only has 15 keys. IDK why, but I've changed it to 16